integreat_cms/cms/utils/pdf_utils.py

```diff
@@ -110,10 +110,22 @@ def generate_pdf(region, language_slug, pages):
         html = get_template("pages/page_pdf.html").render(context)
         # Save empty file
         pdf_storage.save(filename, ContentFile(""))
+
+        # Get fixed version of default pdf styling (see https://github.com/digitalfabrik/integreat-cms/issues/1537)
+        css_file_path = os.path.join(
+            os.path.dirname(__file__), "../../static/src/css/pdf_default_styles.css"
```

Comment:
Maybe `settings.STATIC_ROOT` is a better source for accessing the `static` directory?

Reply:
Ah, right. Will do, thanks.

Reply:
Unfortunately, this isn't as straightforward as it might seem: `STATIC_ROOT` is only used in production mode and we probably want to fix the bug in dev mode as well.

The universal alternative to get the path of static files in both production and develop modee would be 
```
from django.contrib.staticfiles import finders

css_file_path = finders.find('css/pdf_default_styles.css')
```
However, this doesn't work out of the box because we have an intermediary bundling step of the static files via webpack - so in order for this to work, we would have to create a new file `pdf_default_styles.ts` which imports the css file and is then referenced in the [webpack config](https://github.com/digitalfabrik/integreat-cms/blob/develop/webpack.config.js).

So in this case, I guess I would prefer it if we would bypass the whole problem and just modify the existing default css:

```
from xhtml2pdf.default import DEFAULT_CSS

# Remove first occurrence of default html background color style
fixed_css = DEFAULT_CSS.replace("background-color: transparent;", "", 1)
```

Reply:
Oh, or another solution which would be a bit more robust against changes to the upstream repo would be to add the lines
```
html {
    background-color: inherit;
}
```
to the file [integreat_cms/static/src/css/pdf_page_export.css](https://github.com/digitalfabrik/integreat-cms/blob/develop/integreat_cms/static/src/css/pdf_page_export.css)

Reply:
@timoludwig ah, the first suggestion is much nicer, agreed.

The second one (resetting the affected style) unfortunately doesn't work. I initially tried to solve the problem that way, but having *any* `background-color: ` in the `html` style rules breaks the icons - `inherit`, `initial`, `unset`, rgb colors, even invalid values. That's why I think there's some underlying issue, not necessarily with the CSS itself (otherwise the `transparent` background color should be fine as well)

Reply:
Ah okay, I tried it earlier and it seemed to work... 
One potential pitfall: our PDFs are cached in the src/integreat_cms/pdf directory, so maybe the broken version of the pdf was still cached and you didn't notice that the fix works? Could you delete the contents of that directory and try again? Sorry if I mixed something up and this does indeed not work :innocent: